### ISSUE TITLE
Implement Project page sorting

### ISSUE THREAD

**Issue by arkid15r** [OWASP/Nest#73]
For the new frontend add ability to sort projects by starts/contributors/forks/other metadata attributes.

Based on the user's feedback:
> and also to sort by the number of contributors/stars

**Comment by tsu-ki** [comment]
can you please elaborate on what needs to be done here? 
I apologise if this seems like a menial task but since it's my first issue, I want to be sure that I know correctly what I'm working with here.

**Comment by arkid15r** [comment]
This one may be not that easy. It requires both backend and frontend logic implementation. It seems the backend part will require additional index setup:
  - https://www.algolia.com/doc/guides/managing-results/refine-results/sorting/
  - https://www.algolia.com/doc/guides/managing-results/refine-results/sorting/how-to/sort-by-attribute/

I didn't add any details and was going to work on it myself. But if you feel it's something that you can handle let do that -- I'd be happy to guide you and help with any questions. 

@tsu-ki let me know what you think and I'll try to come up w/ more detailed description.

**Comment by tsu-ki** [comment]
sure, i'll quickly go through the code and the guide you've shared. thank you for helping me out

**Comment by arkid15r** [comment]
Hey @tsu-ki any progress on this? Feel free to reach out to me here or in our Slack channel (DMs work too) in case you need help with this one.

**Comment by arkid15r** [comment]
re-assigning after discussing it in Slack

**Comment by Rajgupta36** [comment]
@arkid15r , i can implement it in frontend.

**Comment by arkid15r** [comment]
> @arkid15r , i can implement it in frontend.

Please read [this comment](https://github.com/OWASP/Nest/issues/73#issuecomment-2466876161). We use index engine to produce relevant search results so I doubt I can be done on the frontend.

**Comment by Rajgupta36** [comment]
> > @arkid15r , i can implement it in frontend.
> 
> Please read [this comment](https://github.com/OWASP/Nest/issues/73#issuecomment-2466876161). We use index engine to produce relevant search results so I doubt I can be done on the frontend.

Yes, after getting the results, we can sort them on the frontend. It's quite easy compared to making changes in the API call.

**Comment by arkid15r** [comment]
My point is that search/ranking/sorting can be done on the index engine side only so the frontend sorting would be limited to one page only that doesn't add a lot of value. 
Let me know if I'm missing something

**Comment by Rajgupta36** [comment]
You're right. Frontend sorting is limited to one page and isn't good for large datasets. However, we could use frontend sorting for single pages to ensure better user experience and reduce API calls, while relying on backend sorting to handle larger datasets across pages. what're your thoughts on this approach!
